File: src/main/java/com/back/nbe9112team06/global/security/SecurityUser.java
package com.back.nbe9112team06.global.security;

import lombok.Getter;
import org.springframework.security.core.userdetails.User;

import java.util.List;

@Getter
public class SecurityUser extends User {

    private final int id;
    private final String nickname;

    public SecurityUser(int id, String nickname) {
        super(
                String.valueOf(id),
                "{noop}",
                List.of()
        );
        this.id = id;
        this.nickname = nickname;
    }
}

File: src/main/kotlin/com/back/nbe9112team06/global/security/SecurityUser.kt
package com.back.nbe9112team06.global.security

import org.springframework.security.core.userdetails.User

// TODO java코드에서는 생성자를 따로 만들었지만 kotlin에서는 이미 존재한다고 경고 추후 문제 생길 가능성 OK
class SecurityUser(
    val id: Int,
    val nickname: String
) : User(
    id.toString(),
    "{noop}",
    emptyList()
)